extensions/data-plane-transfer/data-plane-transfer-sync/src/main/java/org/eclipse/dataspaceconnector/transfer/dataplane/sync/flow/ProviderDataPlaneProxyDataFlowController.java

```diff
@@ -0,0 +1,73 @@
+/*
+ *  Copyright (c) 2022 Amadeus
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Amadeus - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.transfer.dataplane.sync.flow;
+
+import org.eclipse.dataspaceconnector.spi.asset.DataAddressResolver;
+import org.eclipse.dataspaceconnector.spi.message.RemoteMessageDispatcherRegistry;
+import org.eclipse.dataspaceconnector.spi.response.ResponseStatus;
+import org.eclipse.dataspaceconnector.spi.transfer.flow.DataFlowController;
+import org.eclipse.dataspaceconnector.spi.transfer.flow.DataFlowInitiateResult;
+import org.eclipse.dataspaceconnector.spi.types.domain.edr.EndpointDataReferenceMessage;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataRequest;
+import org.eclipse.dataspaceconnector.transfer.dataplane.spi.security.DataPlaneProxyManager;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashMap;
+
+public class ProviderDataPlaneProxyDataFlowController implements DataFlowController {
+
+    private static final String HTTP_PROXY_TYPE = "HttpProxy";
+
+    private final String connectorId;
+    private final RemoteMessageDispatcherRegistry dispatcherRegistry;
+    private final DataAddressResolver resolver;
+    private final DataPlaneProxyManager proxyManager;
+
+    public ProviderDataPlaneProxyDataFlowController(String connectorId,
+                                                    RemoteMessageDispatcherRegistry dispatcherRegistry,
+                                                    DataAddressResolver resolver,
+                                                    DataPlaneProxyManager proxyManager) {
+        this.connectorId = connectorId;
+        this.dispatcherRegistry = dispatcherRegistry;
+        this.resolver = resolver;
+        this.proxyManager = proxyManager;
+    }
+
+    @Override
+    public boolean canHandle(DataRequest dataRequest) {
+        return HTTP_PROXY_TYPE.equals(dataRequest.getDestinationType());
+    }
+
+    @Override
+    public @NotNull DataFlowInitiateResult initiateFlow(DataRequest dataRequest) {
+        var address = resolver.resolveForAsset(dataRequest.getAssetId());
+        var proxyCreationResult = proxyManager.createProxy(dataRequest.getId(), address, dataRequest.getContractId(), new HashMap<>());
```

Comment:
Probably `CONTRACT_ID` property needs to be added

Reply:
yeap done